db/repository.py

```
@@ -242,6 +242,20 @@ async def get_by_user_id(cls, user_id: UUID) -> UserProfile | None:
 
+    @classmethod
+    async def debug_lookup_by_name(cls, name: str) -> dict | None:
+        """Debug helper for looking up a profile by user name."""
+        async with Database.acquire() as conn:
+            row = await conn.fetchrow(
+                f"""
+                SELECT up.*, u.name
+                FROM user_profile up
+                JOIN "user" u ON u.id = up.user_id
+                WHERE u.name = '{name}'
+                """
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)